sdk/azcore/arm/internal/resource/resource_identifier.go

```diff
@@ -110,6 +110,21 @@ func (id *ResourceID) String() string {
 	return id.stringValue
 }
 
+// MarshalText returns a textual representation of the ResourceID
+func (id *ResourceID) MarshalText() ([]byte, error) {
```

Comment:
Usually the receiver is by value for `Marshal*` methods. Is there a reason why it's a pointer here?

Reply:
Mainly because `ParseResourceID` allocates a new `ResourceID`, and I was using [url.URL](https://pkg.go.dev/net/url#URL) as a guide.  Its [MarshalBinary](https://pkg.go.dev/net/url#URL.MarshalBinary) and [UnmarshalBinary](https://pkg.go.dev/net/url#URL.UnmarshalBinary) methods both take pointers.

It's less important for `MarshalText` but I think its the only way to get `UnmarshalText` to work without doing a field-by-field assignment.